webapp/pages/profile/_id/_slug.vue

```diff
@@ -348,6 +379,19 @@ export default {
         return []
       }
       return this.uniq(this.user.contributions.filter(post => !post.deleted))
+    },
+    socialMediaLinks() {
+      const { socialMedia = [] } = this.user
+      return socialMedia.map(socialMedia => {
+        const { url } = socialMedia
+        const matches = url.match(
+          /^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:\/\n?]+)/g
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T15:43:46Z" title="Friday, April 12th 2019, 5:43:46 pm +02:00">Apr 12, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -348,6 +379,19 @@ export default {
         return []
       }
       return this.uniq(this.user.contributions.filter(post => !post.deleted))
+    },
+    socialMediaLinks() {
+      const { socialMedia = [] } = this.user
+      return socialMedia.map(socialMedia => {
+        const { url } = socialMedia
+        const matches = url.match(
+          /^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:\/\n?]+)/g
```


I wonder if we could avoid to roll our own URL regex. How about using NodeJS `url` module? https://stackoverflow.com/questions/30931079/validating-a-url-in-node-js

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-12T16:15:53Z" title="Friday, April 12th 2019, 6:15:53 pm +02:00">Apr 12, 2019</time>_

---


that is an interesting idea.. I'm ok to do that, I just don't know how other people feel about adding an npm package for just that, but maybe I am reading a different answer on that SO post you posted :point_up:

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T16:19:08Z" title="Friday, April 12th 2019, 6:19:08 pm +02:00">Apr 12, 2019</time>_

---


[nodejs url](https://nodejs.org/api/url.html) is a package which is available by default on NodeJS. Maybe we have to polyfill it on the browser (if it's not already polyfilled by Vue&friends).